test/integration/confirmations/transactions/contract-deployment.test.tsx

```diff
@@ -0,0 +1,408 @@
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+  callBackgroundMethod: jest.fn(),
+}));
+
+const mockedBackgroundConnection = jest.mocked(backgroundConnection);
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+export const pendingTransactionId = '48a75190-45ca-11ef-9001-f3886ec2397c';
+export const pendingTransactionTime = new Date().getTime();
+
+const getMetaMaskStateWithUnapprovedContractDeployment = ({
+  accountAddress,
+  showConfirmationAdvancedDetails = false,
+}: {
+  accountAddress: string;
+  showConfirmationAdvancedDetails?: boolean;
+}) => {
+  return {
+    ...mockMetaMaskState,
+    preferences: {
+      ...mockMetaMaskState.preferences,
+      redesignedConfirmationsEnabled: true,
+      showConfirmationAdvancedDetails,
+    },
+    nextNonce: '8',
+    currencyRates: {
+      SepoliaETH: {
+        conversionDate: 1721392020.645,
+        conversionRate: 3404.13,
+        usdConversionRate: 3404.13,
+      },
+      ETH: {
+        conversionDate: 1721393858.083,
+        conversionRate: 3414.67,
+        usdConversionRate: 3414.67,
+      },
+    },
+    currentCurrency: 'usd',
+    pendingApprovals: {
+      [pendingTransactionId]: {
+        id: pendingTransactionId,
+        origin: 'local:http://localhost:8086/',
+        time: pendingTransactionTime,
+        type: ApprovalType.Transaction,
+        requestData: {
+          txId: pendingTransactionId,
+        },
+        requestState: null,
+        expectsResult: false,
+      },
+    },
+    pendingApprovalCount: 1,
+    knownMethodData: {
+      '0x3b4b1381': {
+        name: 'Mint NFTs',
+        params: [
+          {
+            type: 'uint256',
+          },
+        ],
+      },
+    },
+    transactions: [
+      getUnapprovedContractDeploymentTransaction(
+        accountAddress,
+        pendingTransactionId,
+        pendingTransactionTime,
+      ),
+    ],
+  };
+};
+
+const advancedDetailsMockedRequests = {
+  getGasFeeTimeEstimate: {
+    lowerTimeBound: new Date().getTime(),
+    upperTimeBound: new Date().getTime(),
+  },
+  getNextNonce: '9',
+  decodeTransactionData: {
+    data: [
+      {
+        name: 'mintNFTs',
+        params: [
+          {
+            name: 'numberOfTokens',
+            type: 'uint256',
+            value: 1,
+          },
+        ],
+      },
+    ],
+    source: 'Sourcify',
+  },
+};
+
+const setupSubmitRequestToBackgroundMocks = (
+  mockRequests?: Record<string, unknown>,
+) => {
+  mockedBackgroundConnection.submitRequestToBackground.mockImplementation(
+    createMockImplementation({
+      ...advancedDetailsMockedRequests,
+      ...(mockRequests ?? {}),
+    }),
+  );
+};
+
+describe('Contract Deployment Confirmation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupSubmitRequestToBackgroundMocks();
+    const MINT_NFT_HEX_SIG = '0x3b4b1381';
+    mock4byte(MINT_NFT_HEX_SIG);
```

Comment:
This is a contract deployment.. should we mock4byte to provide a proper response? For contract deployment I am not sure what are calling 4byte with...
Maybe we need to better setup the mocks for 4byte? 🤔

Reply:
In [this commit](https://github.com/MetaMask/metamask-extension/pull/27259/commits/ea38f8945a7d7f40c95a9e1153bf19323c60615d), I changed the mocked 4 bytes method and signature to deposit, which is the one used in our e2e tests.